flash/vision/classification/data.py

```diff
@@ -277,6 +243,11 @@ def _check_transforms(transform: Dict[str, Union[nn.Module, Callable]]) -> Dict[
                 "Transform should be a dict. "
                 f"Here are the available keys for your transforms: {DataPipeline.PREPROCESS_FUNCS}."
             )
+        if "per_batch_transform" in transform and "per_sample_transform_on_device" in transform:
+            raise MisconfigurationException(
+                f'{transform}: `per_batch_transform` and `per_sample_transform_on_device` '
+                f'are mutual exclusive.'
+            )
```

Comment:
not really sure if we should constrain this. I'm thinking about a use case where you would like to perform an expensive transformation on device (e.g big image) + crop/resize AND later continue with other batched transforms.

Reply:
We don't have a uncollate in the preprocess for now as it would be pretty expensive, therefore those operations are mutually exclusive.

Reply:
We definitely need to constrain this, since otherwise we have to uncollate again and this is not desired.

however you can still apply batched transforms on the device after the per sample transforms on device. just batched transforms within the loader workers and sample transforms on gpu are mutally exclusive since we need to collate within the workers for batched transforms there.